dbos/system_database.py

```diff
@@ -1010,3 +1017,48 @@ def _is_buffers_empty(self) -> bool:
             len(self._workflow_status_buffer) == 0
             and len(self._workflow_inputs_buffer) == 0
         )
+
+    def enqueue(self, workflow_id: str, queue_name: str) -> None:
+        with self.engine.begin() as c:
+            c.execute(
+                pg.insert(SystemSchema.job_queue)
+                .values(
+                    workflow_uuid=workflow_id,
+                    queue_name=queue_name,
+                )
+                .on_conflict_do_nothing()
+            )
+
+    def dequeue(self, queue_name: str, concurrency: Optional[int]) -> List[str]:
```

Comment:
I see, dequeue doesn't really remove a workflow from the queue. Instead, it puts a workflow to a pending state. Can we change this name to something more descriptive? Or add a comment to this function to make it clear...